demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,618 @@
+
+        if( jsonValueLength > JOBS_DEMO_MAX_ID_LENGTH )
+        {
+            keyFound = false;
+            IotLogError( "Job ID is too long." );
+        }
+        else
+        {
+            memcpy( jobId, pJsonValue, jsonValueLength );
+            jobIdLength = jsonValueLength;
+        }
+    }
+    else
+    {
+        IotLogError( "Failed to parse JobID from Notify Next." );
+        IotLogError( "document: %.*s", pCallbackInfo->u.callback.documentLength, pCallbackInfo->u.callback.pDocument );
+    }
+
+    /* Get the Job Document */
+    if( keyFound )
+    {
+        docKeyFound = GET_DOC( &pJsonValue, &jsonValueLength );
+    }
+
+    if( docKeyFound )
+    {
+        if( jsonValueLength > JOBS_DEMO_MAX_JOB_DOC_LENGTH )
+        {
+            keyFound = false;
+            IotLogError( "Job document is too long." );
+        }
+        else
+        {
+            memcpy( jobDoc, pJsonValue, jsonValueLength );
+            jobDocLength = jsonValueLength;
+        }
+
+        /* Execute the demo */
+        status = _executeDemo( pCallbackInfo->mqttConnection,
+                               pCallbackInfo->pThingName,
+                               pCallbackInfo->thingNameLength,
+                               jobId,
+                               jobIdLength,
+                               jobDoc,
+                               jobDocLength );
+    }
+    else
+    {
+        if( keyFound && !docKeyFound )
+        {
+            IotLogError( "Failed to parse Job Document from Notify Next." );
+        }
+    }
+
+#undef GET_KEY
+#undef GET_DOC
+}
+
+/**
+ * @brief Clean up the Jobs demo.
+ *
+ * @note Must not be called if `_initDemo` is not called or does not succeed.
+ */
+static void _cleanupDemo( void )
+{
+    AwsIotJobs_Cleanup();
+    IotMqtt_Cleanup();
+}
+
+/**
+ * @brief Establish a new connection to the MQTT server for the Jobs demo.
+ *
+ * @param[in] pIdentifier NULL-terminated MQTT client identifier. The Jobs
+ * demo will use the Thing Name as the client identifier.
+ * @param[in] pNetworkServerInfo Passed to the MQTT connect function when
+ * establishing the MQTT connection.
+ * @param[in] pNetworkCredentialInfo Passed to the MQTT connect function when
+ * establishing the MQTT connection.
+ * @param[in] pNetworkInterface The network interface to use for this demo.
+ * @param[out] pMqttConnection Set to the handle to the new MQTT connection.
+ *
+ * @return `EXIT_SUCCESS` if the connection is successfully established; `EXIT_FAILURE`
+ * otherwise.
+ */
+static int32_t _establishMqttConnection( const char * pIdentifier,
+                                         void * pNetworkServerInfo,
+                                         void * pNetworkCredentialInfo,
+                                         const IotNetworkInterface_t * pNetworkInterface,
+                                         IotMqttConnection_t * const pMqttConnection )
+{
+    int32_t status = EXIT_SUCCESS;
```

Comment:
int type for status

Reply:
I've changed it, but `EXIT_SUCCESS` is a signed literal.